docs/model_system/model_system.md

```diff
@@ -1,4 +1,209 @@
+
+```python
+model_system = ModelSystem()
+# One physical system (e.g., silicon crystal)
+
+# Multiple cells for different representations:
+model_system.cell[0]  # Original/input cell
+model_system.cell[1]  # Primitive cell  
+model_system.cell[2]  # Conventional cell
+```
+
+<!-- @ndaelman-hu: am not completely sold on the placement of `cell`, as all the crystallographic examples tie in with `Symmetry` -->
+
+  ## Subsystems and Hierarchical Organization
+
+  `ModelSystem` supports **hierarchical subsystem organization** beyond just multiple cell representations. This enables modeling of complex multi-component systems like heterostructures, surfaces with adsorbates, or solutions, monomers within polymers, elemental distribution.
+  These subsets do **not** have to form a partition.
+
+  ### Key Concepts:
+  - **Root system**: Contains all atomic positions and the complete structural data
+  - **Subsystems**: Reference subsets of atoms via `particle_indices` without duplicating geometric data
+  - **Hierarchical nesting**: Subsystems can contain their own subsystems (e.g., device → component → molecule)
+
+
+All subsystems leverage previously registered geometric data by referring directly to the **top-level node**, i.e. the **complete system**.
+This allows for single-access lookups rather than recursive traversals.
+
+  ```python
+  # Example: Heterostructure modeling
+  heterostructure = ModelSystem(name='Si/GaAs', is_representative=True)
+  heterostructure.positions = [...]  # All atomic positions
+
+  # Subsystem 1: Silicon layer
+  si_layer = ModelSystem(name='Si_layer', particle_indices=[0, 1, 2, 3])
+  heterostructure.sub_systems.append(si_layer)
+
+  # Subsystem 2: GaAs layer
+  gaas_layer = ModelSystem(name='GaAs_layer', particle_indices=[4, 5, 6, 7, 8, 9])
+  heterostructure.sub_systems.append(gaas_layer)
+```
+
+
+## Structure Representations via `Cell` Sections
+
+### Cell Types and Their Purpose
+
+1. **Original Cell** (`type='original'`)
+   - The cell as provided in the input/calculation
+   - May be any arbitrary choice of unit cell
+   - Contains the lattice vectors used in the actual computation
+
+2. **Primitive Cell** (`type='primitive'`)  
+   - Smallest possible unit cell that tiles the crystal
+   - Contains minimum number of lattice points
+   - Used for certain calculations (e.g., band structures)
+
+3. **Conventional Cell** (`type='conventional'`)
+   - Standardized cell following International Tables conventions
+   - Used for symmetry analysis and crystallographic databases
+   - **Typically used to define symmetry operators:** ["x,y,z", "-x,y,-z", "x+1/2,y+1/2,z", "-x+1/2,y+1/2,-z"] (conventional) vs ["x,y,z", "-x,-y,z", "-x,y,-z", "x,-y,-z"] (primitive).
```

Comment:
It should be maybe noted here that primitive and conventional systems may not directly correspond to the original cell: they will have different densities and "idealized" lattice parameters to match space group that they were matched with certain precision.

Reply:
I will add this as a clarification to the description.

> they will have different densities

Not sure what you mean with densities here: mass, electronic? The former stays the same, the latter changes exactly like the geometry. However, k-samplings should change too, and this is not acknowledged (or addressed) anywhere.

Reply:
Ow, upon re-read, i think I understand it now. You're referring to deviations in the cell volume due to the symmetrization accuracy constraint.